developers/weaviate/configuration/rbac/manage-users.mdx

```diff
@@ -0,0 +1,476 @@
+  <TabItem value="py" label="Python Client v4">
+  <FilteredTextBlock
+    text={UserPyCode}
+    startMarker="# START DeleteUser"
+    endMarker="# END DeleteUser"
+    language="py"
+  />
+  </TabItem>
+  <TabItem value="js" label="JS/TS Client v3">
+
+```ts
+// TS support coming soon
+```
+
+  </TabItem>
+  <TabItem value="go" label="Go">
+
+```go
+// Go support coming soon
+```
+
+  </TabItem>
+  <TabItem value="java" label="Java">
+
+```java
+// Java support coming soon
+```
+
+  </TabItem>
+</Tabs>
+
+### List all users
+
+This example shows how to get a list of all the users in Weaviate.
+
+<Tabs groupId="languages">
+  <TabItem value="py" label="Python Client v4">
+  <FilteredTextBlock
+    text={UserPyCode}
+    startMarker="# START ListAllUsers"
+    endMarker="# END ListAllUsers"
+    language="py"
+  />
+  </TabItem>
+  <TabItem value="js" label="JS/TS Client v3">
+
+```ts
+// TS support coming soon
+```
+
+  </TabItem>
+  <TabItem value="go" label="Go">
+
+```go
+// Go support coming soon
+```
+
+  </TabItem>
+  <TabItem value="java" label="Java">
+
+```java
+// Java support coming soon
+```
+
+  </TabItem>
+</Tabs>
+
+<details>
+  <summary>Example results</summary>
+
+```text
+[
+  UserDB(user_id='custom-user', role_names=['viewer', 'testRole'], user_type=<UserTypes.DB_DYNAMIC: 'db_user'>, active=True), 
+  UserDB(user_id='root-user', role_names=['root'], user_type=<UserTypes.DB_STATIC: 'db_env_user'>, active=True)
+]
+```
+
+</details>
+
+:::info User types in Weaviate
+<!-- NOTE: left out AUTHORIZATION_ADMINLIST_USERS to dissuade usage -->
+<!-- TODO[g-despot]: Check if DB_DYNAMIC is renamed -->
+Under the hood, Weaviate differentiates three types of users:
+- `DB_DYNAMIC`: Database users that can be fully managed through the API.
+- `DB_STATIC`: Database users that are defined through the `AUTHENTICATION_APIKEY_USERS` environment variable and can only be updated through this variable and by restarting the Weaviate instance.
+- `OIDC`: Users that can only be created/deleted through the external OIDC service.
+
+:::
+
+### Rotate user API key
+
+This example updates (rotates) the API key for `custom-user`.
+
+<Tabs groupId="languages">
+  <TabItem value="py" label="Python Client v4">
+  <FilteredTextBlock
+    text={UserPyCode}
+    startMarker="# START RotateApiKey"
+    endMarker="# END RotateApiKey"
+    language="py"
+  />
+  </TabItem>
+  <TabItem value="js" label="JS/TS Client v3">
+
+```ts
+// TS support coming soon
+```
+
+  </TabItem>
+  <TabItem value="go" label="Go">
+
+```go
+// Go support coming soon
+```
+
+  </TabItem>
+  <TabItem value="java" label="Java">
+
+```java
+// Java support coming soon
+```
+
+  </TabItem>
+</Tabs>
+
+<details>
+  <summary>Example results</summary>
+
+```text
+SSs3WGVFbUxMVFhlOEsxVVMrQVBzM1VhQTJIM2xXWngwY01HaXFYVnM1az1fMXlDUEhUNjhSMlNtazdHcV92MjAw
+```
+
+</details>
+
+## User permissions management <i class="fas fa-user-lock"></i> {#user-permissions-management}
```

Comment:
I think I might want to know what subset of users this applies to (e.g. DB + DB_ENV users)